src/backend/Order/Actions/CreateOrderActions/CreateOrderAddAddress/CreateOrderAddAddressService.cs

```diff
@@ -0,0 +1,112 @@
+            _logger = logger;
+            _hashids = hashids;
+            _currentUserService = currentUserService;
+            _getCurrentUserService = getCurrentUserService;
+        }
+
+        public async Task<CreateOrderAddAddressResponse> AddAddress(CreateOrderAddAddressPayload payload)
+        {
+            if (!ValidatePayload(payload))
+            {
+                _logger.LogError("Invalid payload");
+                throw new ClientInputErrorException("Sorry, we couldn't fetch your new address. Please contact service administrator.");
+            }
+
+            var auth0UserInfo = await _currentUserService.GetUserInfo();
+            var currentUser = await _getCurrentUserService.Execute(auth0UserInfo);
+
+            var country = _appDbContext.Countries.SingleOrDefaultAsync(c => c.EnglishName == payload.CountryName);
+            if (country.Result == null)
+            {
+                _logger.LogError("Couldn't find country by it's EnglishName");
+                throw new ItemNotFoundException("Cannot find given country name in database. Please provide valid country name.");
+            }
+
+            var newAddress = new Address()
+            {
+                Country = country.Result,
+                PostalCode = payload.PostalCode,
+                City = payload.City,
+                Street = payload.Street,
+                StreetNumber = payload.StreetNumber,
+                Apartment = payload.Apartment,
+                GpsLatitude = payload.GpsLatitude,
+                GpsLongitude = payload.GpsLongitude,
+                UpdateUserId = UserSeeder.ServiceUser.Id,
+                UpdatedAt = DateTime.UtcNow,
+                Status = DbEntityStatus.Active
+            };
+            await _appDbContext.AddAsync(newAddress);
+
+            var newUserAddress = new UserAddress()
+            {
+                User = currentUser,
+                Address = newAddress,
+                IsDeletedByUser = false,
+                UpdateUserId = UserSeeder.ServiceUser.Id,
+                UpdatedAt = DateTime.UtcNow,
+                Status = DbEntityStatus.Active
+            };
+            await _appDbContext.AddAsync(newUserAddress);
+
+            await _appDbContext.SaveChangesAsync();
+
+            var response = new CreateOrderAddAddressResponse()
+            {
+                Id = _hashids.EncodeLong(newAddress.Id),
+                CountryName = newAddress.Country!.EnglishName,
+                PostalCode = newAddress.PostalCode,
+                City = newAddress.City,
+                Street = newAddress.Street,
+                StreetNumber = newAddress.StreetNumber,
+                Apartment = newAddress.Apartment,
+                GpsLatitude = newAddress.GpsLatitude,
+                GpsLongitude = newAddress.GpsLongitude
+            };
+
+            return response;
+        }
+
+        private static bool ValidatePayload(CreateOrderAddAddressPayload payload)
```

Comment:
Do wywalenia.
Skoro mamy FluentValidation to go używaj. Sprawdź jak robiona jest walidacja w innych klasach.

Reply:
wywalone, poprawione